src/parser/core/modules/Combos.js

```diff
@@ -0,0 +1,118 @@
+// If you can make it through this entire file without hitting semantic saturation of the word "combo", hats off to you. IT DOESN'T LOOK REAL ANYMORE.
+
+import React, {Fragment} from 'react'
+
+import {getAction} from 'data/ACTIONS'
+import Module from 'parser/core/Module'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+const NO_COMBO = -1
+const GCD_TIMEOUT_MILLIS = 12000
+
+export default class Combos extends Module {
+	static handle = 'combos'
+	static dependencies = [
+		'suggestions',
+	]
+
+	// This should be redefined by subclassing modules; the default is the basic 'Attack' icon
+	static suggestionIcon = 'https://secure.xivdb.com/img/game/000000/000405.png'
+
+	_lastAction = NO_COMBO
+	_lastGcdTime = this.parser.fight.start_time
+	_brokenComboCount = 0
+	_uncomboedGcdCount = 0
+
+	_comboBreakers = []
+	_uncomboedGcds = []
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {by: 'player'}, this._onCast) // TODO - Can we filter to onGcd=true? That would be neat.
```

Comment:
Not really, it's not part of the event data. Maybe something to look into normalising in but don't hold your breath.

Reply:
Eh, not really worth normalising. I'll delete the TODO.